src/components/Dropdown/Dropdown.tsx

```diff
@@ -76,14 +78,35 @@ export const Dropdown = ({
 
   const classes = useMemo(() => classNames(styles.dropdownWrapper, uniqueClassName), [uniqueClassName])
 
+  const onOpenChangeInternal = useCallback(
+    (open: boolean, info?: {source: 'trigger'| 'menu'}) => {
+      if (!onOpenChange) {
+        return
+      }
+      switch (info?.source) {
```

Comment:
I'd simplify this code section and make it so the `onOpenChange` gets called only once.

The switch case could be substituted by a simple map, and something like this should do the trick:

```typescript
// outside the component
type antdTriggerSource = 'trigger'|'menu'
const antdSourceMap: Record<antdTriggerSource,OpenChangeInfoSource> = {
'trigger': OpenChangeInfoSource.Trigger,
'menu': OpenChangeInfoSource.Menu,
}

// here
const source = antdSourceMap[info?.source]
onOpenChange(open, source ? {source} : undefined)
```

Reply:
`antdSourceMap[info?.source]` leads to a type error due to the fact that `source` cannot be `undefined`, and neither the object key can be.
I had to leave the check at runtime, still simplifying a little bit the code

Reply:
I actually managed to remove the check, it should be acceptable now